codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -321,6 +321,21 @@ class ServerServiceGeneratorV2(
                         router: self.router.map(|s| s.layer(layer))
                     }
                 }
+
+                /// Applies [`Route::new`](#{SmithyHttpServer}::routing::Route::new) to all routes.
+                ///
+                /// This has the effect of erasing all types accumulated via [`layer`].
+                pub fn boxed<B>(self) -> $serviceName<#{SmithyHttpServer}::routing::Route<B>>
```

Comment:
Couldn't this method go in an extension trait in a runtime crate?
Bringing this up because I feel that less generated code is usually better.

Reply:
I actually believe the opposite.. generated code is easier to manage and release than runtime code..

Reply:
If this was a standalone function I think it might fit better in the `rust-runtime` crates but given that this is a small method on a generated struct I think it makes sense to leave it here for now. If we did this via extension trait the customer would have to import the trait to use the method which may be confusing.

It is possible though: we'd need to add a base trait with the `layer` method on it, impl that onto `PokemonService` via codegen, and then include this `boxed` method extension to that trait.

Maybe we should draft out a heuristic to aid in this kind of decision making.

Reply:
Yeah, we should figure out guidelines to decide whether to put stuff in runtime vs codegen!

Reply:
In this case I'd lean towards putting this method in the code generated `struct`. It's a better developer experience than having to import a single-use extension trait just to call it.